redis/src/main/scala/zio/redis/RedisUri.scala

```diff
@@ -16,8 +16,17 @@
 
 package zio.redis
 
-final case class RedisConfig(host: String, port: Int)
+final case class RedisUri(host: String, port: Int) {
+  override def toString: String = s"$host:$port"
+}
+
+object RedisUri {
+  lazy val Default: RedisUri = RedisUri("localhost", 6379)
 
-object RedisConfig {
-  lazy val Default: RedisConfig = RedisConfig("localhost", 6379)
+  def apply(hostAndPort: String): RedisUri = {
+    val splitting = hostAndPort.split(':')
+    val host      = splitting(0)
+    val port      = splitting(1).toInt
+    RedisUri(host, port)
```

Comment:
Why is this necessary?

Reply:
Cause slot errors returns in this format where i should redirect my request